retry.go

```diff
@@ -166,8 +184,12 @@ func (r *Retry) Timeout(to time.Duration) *Retry {
 
 	deadline := time.Now().Add(to)
 
-	r.appendPreCondition(func() bool {
-		return time.Now().Before(deadline)
+	r.appendPreCondition(func(err error) error {
+		now := time.Now()
+		if now.After(deadline) || now.Equal(deadline) {
```

Comment:
is this `now.Equal` really necessary?

Reply:
I think `!now.Before(deadline)` would be a better convention, but I don't think the case where the nanosecond matches up is worth considering ;)

Reply:
!now.Bofore(deadline) is what I used to use too but I stopped because too clever. I think its more clear when the condition is a positive. Its not a big deal when the nanosecond matches up but its more a semantic thing for me. Like deadline is defined as the time at which the timeout occurs, not the time after which the timeout occurs.